### PR TITLE
[FIXED] Pull requests: don't send 408 when request expires

### PR DESCRIPTION
**Pull request by kozlovic**
When expiring requests, the server would send 408 if interest was
still present, which can happen for pull subscribe implementations
that maintain interest for the duration of the pull subscription.

Let's keep the 408 for when a request is "force expired", that
is, a request was removed from the queue because it queue was
full but interest is still found.

Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

File: server/jetstream_test.go [hunk 1]
```diff
@@ -1853,13 +1853,15 @@ func TestJetStreamWorkQueueRequest(t *testing.T) {
 			// Send a few more messages. These should not be delivered to the sub.
 			sendStreamMsg(t, nc, "foo", "Hello World!")
 			sendStreamMsg(t, nc, "bar", "Hello World!")
-			// We will have an alert here.
+			time.Sleep(10 * time.Millisecond)
```
Comment (derekcollison) [inline server/jetstream_test.go:1856]: Would make at least 100 these days.

**Review by derekcollison — APPROVED** [review]
LGTM - Might bump one sleep.

File: server/jetstream_test.go (hunk 1 above)
Comment (kozlovic) [reply server/jetstream_test.go:1856]: We sleep above past the expiration time. This sleep was to make sure that we would not start delivering, but I hear you. Will increase.

Comment (kozlovic) [reply server/jetstream_test.go]: Increased the expiration timeout anyway (but still "small") and bumped server version.

**Review by derekcollison — APPROVED** [review]
LGTM

**Comment by ripienaar** [comment]
Very glad to see this removed LGTM

**Comment by kozlovic** [comment]
@ripienaar It is still sent (as of now) if the request is "force expired", that is, the server could not remove (due to expiration) a single request and therefore evicts the first from the queue. In that case, if there is interest, server will still send 408, which I would argue is wrong "status/description", since in this case it is not a timeout but an eviction or some other error, but that would definitively break clients if we introduce a new code there.

**Comment by ripienaar** [comment]
Yes, my moan about them was the ones that came at some random future time and not at the point of expiration timer expiring (the lazy ones).  So happy to see those specific ones go away.

**Pull request merged**